jerry-core/vm/opcodes.c

```diff
@@ -716,14 +762,15 @@ opfunc_resume_executable_object (vm_executable_object_t *executable_object_p, /*
   JERRY_CONTEXT (vm_top_context_p) = executable_object_p->frame_ctx.prev_context_p;
 
   register_p = VM_GET_REGISTERS (&executable_object_p->frame_ctx);
-
-  while (register_p < register_end_p)
-  {
-    ecma_deref_if_object (*register_p++);
-  }
+  stack_top_p = executable_object_p->frame_ctx.stack_top_p;
 
   if (executable_object_p->frame_ctx.context_depth > 0)
```

Comment:
This `if` and  the `while` after it seems to me used in this form multiple times (with seemingly minimal changes). Would it be possible to extract them into a method?

Reply:
This is only used twice, and I would not combine them because they do opposite things.

Reply:
ohh I see one is `ref` and the other is `deref`. Sorry for the noise I've missed this.